src/pyhf_combine_converter/pyhf_converted_from_datacard.py

```diff
@@ -300,33 +305,41 @@ def addMods(spec: dict):  ##add systematics as modifiers
                         )
 
 
-def toJSON(spec: dict):
-    addChannels(spec)
-    addSamples(spec)
-    addMeasurements(spec)
-    addNormFactor(spec)
-    addMods(spec)
-
+def main():
+    parser = OptionParser()
+    DP.addDatacardParserOptions(parser)
+    parser.add_option(
+        "-O",
+        "--out-file",
+        dest="outfile",
+        default="converted_workspace.json",
+        help="desired name of JSON file",
+    )
+    options, args = parser.parse_args()  # add command line args
 
-def writeFileName(name):
-    with open(name, "w") as file:
-        file.write(json.dumps(spec, indent=2))
-
-if __name__ == "__main__":
-    DC = Datacard()  ##create Datacard object
+    data_card = Datacard()  # create Datacard object
     with open(args[0]) as dc_file:
-        DC = Datacard()
-        DC = DP.parseCard(file=dc_file, options=options)
-   
-
-    channels = [channel for channel in DC.bins]
-    observations = [obs for channel, obs in DC.obs.items()]
-    samples = [sample for sample in DC.processes]
-    exp_values = DC.exp
-    sig = DC.isSignal
-    mods = DC.systs
+        data_card = Datacard()
+        data_card = DP.parseCard(file=dc_file, options=options)
+
+    channels = [channel for channel in data_card.bins]
+    observations = [obs for channel, obs in data_card.obs.items()]
+    samples = [sample for sample in data_card.processes]
+    exp_values = data_card.exp
+    sig = data_card.isSignal
+    mods = data_card.systs
     spec = {"channels": [], "observations": [], "measurements": [], "version": "1.0.0"}
-    toJSON(spec)
-    writeFileName(options.outfile)
 
+    # convert to JSON spec
+    addChannels(spec, data_card, channels, observations)
+    addSamples(spec, data_card, channels, samples, exp_values)
+    addMeasurements(spec, data_card, channels, samples, sig)
+    addNormFactor(spec, data_card, channels, samples, sig)
+    addMods(spec, data_card, channels, samples, exp_values, mods)
```

Comment:
@peterridolfi Before hand everything was in global scope and so functions could access data structures that were not passed in as arguments. In general we want to avoid this, as it makes it easy to accidentally mutate data.

This is just passing in the necessary data structures into the functions, but as I've revised the API a bit here please make sure that the tests that you've run so far work as expected before any of this gets merged. :+1: